app/lib/actions/katello/capsule_content/sync.rb

```diff
@@ -38,13 +38,24 @@ def plan(smart_proxy, options = {})
           affected_repos = affected_repositories(capsule_content, environment, content_view, repository)
           need_updates = repos_needing_updates(capsule_content, affected_repos)
           repository_ids = get_repository_ids(capsule_content, environment, content_view, repository)
+
+          # Create a list of non-puppet repos to sync first, and then sync puppet repos last.
+          # Puppet modules may refer to non-puppet content in the same CV, and thus need to publish last.
+          puppet_repository_ids = []
+          non_puppet_repository_ids = []
+          repository_ids.each do |repo_id|
```

Comment:
Rails provides a [built in way to split arrays](http://api.rubyonrails.org/classes/Array.html#method-i-split) with arbitrary logic if you want to use it:

```rb
puppet_repository_ids, non_puppet_repository_ids = repository_ids.split do |repo_id|
  ...
end
```

Reply:
@akofink Is that appropriate here? looks like that splits on anything matching the block into a 2d array as opposed to creating two groups, and doesn't include the splitting element.

Reply:
Oh, the block sets the delimiter. Hmm... yeah, it's not really useful here after all. :)